Sources/SmokeDynamoDB/AWSDynamoDBCompositePrimaryKeysProjection+DynamoDBKeysProjectionAsync.swift

```diff
@@ -20,91 +20,79 @@ import SmokeAWSCore
 import DynamoDBModel
 import SmokeHTTPClient
 import Logging
-import NIO
 
 /// DynamoDBKeysProjection conformance async functions
 public extension AWSDynamoDBCompositePrimaryKeysProjection {
     
-    func query<AttributesType>(
-            forPartitionKey partitionKey: String,
-            sortKeyCondition: AttributeCondition?) -> EventLoopFuture<[CompositePrimaryKey<AttributesType>]>
-            where AttributesType: PrimaryKeyAttributes {
-        return partialQuery(forPartitionKey: partitionKey,
-                            sortKeyCondition: sortKeyCondition,
-                            exclusiveStartKey: nil)
+    func query<AttributesType>(forPartitionKey partitionKey: String,
+                                      sortKeyCondition: AttributeCondition?) async throws
+    -> [CompositePrimaryKey<AttributesType>] {
+        return try await partialQuery(forPartitionKey: partitionKey,
+                                      sortKeyCondition: sortKeyCondition,
+                                      exclusiveStartKey: nil)
     }
     
     // function to return a future with the results of a query call and all future paginated calls
     private func partialQuery<AttributesType>(
             forPartitionKey partitionKey: String,
             sortKeyCondition: AttributeCondition?,
-            exclusiveStartKey: String?) -> EventLoopFuture<[CompositePrimaryKey<AttributesType>]>
-            where AttributesType: PrimaryKeyAttributes {
-        let queryFuture: EventLoopFuture<([CompositePrimaryKey<AttributesType>], String?)> =
-            query(forPartitionKey: partitionKey,
-                  sortKeyCondition: sortKeyCondition,
-                  limit: nil,
-                  scanIndexForward: true,
-                  exclusiveStartKey: exclusiveStartKey)
+            exclusiveStartKey: String?) async throws
+    -> [CompositePrimaryKey<AttributesType>] {
+        let paginatedItems: ([CompositePrimaryKey<AttributesType>], String?) =
```

Comment:
nit, can we add named parameters to the tuple so the below code is slightly easier to reason about?